storage/client/blob/src/samples/java/blob/FileTransferExample.java

```diff
@@ -0,0 +1,177 @@
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Locale;
+
+/**
+ * This class shows how to upload the file as fast as possible in parallel using the optimized upload API.
+ */
+public class FileTransferExample {
+    private static final String LARGE_TEST_FOLDER = "test-large-files/";
+    public static void main(String[] args) throws Exception {
+
+        /**
+         * From the Azure portal, get your Storage account's name and account key.
+         */
+        String accountName = getAccountName();
+        String accountKey = getAccountKey();
+
+        /**
+         * Use your Storage account's name and key to create a credential object; this is used to access your account.
+         */
+        SharedKeyCredential credential = new SharedKeyCredential(accountName, accountKey);
+
+        /**
+         * From the Azure portal, get your Storage account blob service URL endpoint.
+         * The URL typically looks like this:
+         */
+        String endPoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);
+
+        /**
+         * Create a StorageClient object that wraps the service endpoint, credential and a request pipeline.
+         * Now you can use the storageClient to perform various container and blob operations.
+         */
+        StorageClient storageClient = StorageClient.storageClientBuilder().endpoint(endPoint).credential(credential).buildClient();
+
+
+        /**
+         * This example shows several common operations just to get you started.
+         */
+
+
+        /**
+         * Create a client that references a to-be-created container in your Azure Storage account. This returns a
+         * ContainerClient uses the same endpoint, credential and pipeline from storageClient.
+         * Note that container names require lowercase.
+         */
+        ContainerClient containerClient = storageClient.getContainerClient("myjavacontainerparallelupload" + System.currentTimeMillis());
+
+        /**
+         * Create a container in Storage blob account.
+         */
+        containerClient.create();
+
+        /**
+         * Create a BlockBlobClient object that wraps a blob's endpoint and a default pipeline, the blockBlobClient give us access to upload the file.
+         */
+        String filename = "BigFile.bin";
+        BlockBlobClient blobClient = containerClient.getBlockBlobClient(filename);
+
+        /**
+         * Create the empty uploadFile and downloadFile.
+         */
+        File largeFile = createTempEmptyFile(filename);
+
+        File downloadFile = createTempEmptyFile("downloadFile.bin");
+
+        /**
+         * Generate random things to uploadFile, which makes the file with size of 100MB.
+         */
+        Long fileSize = 100 * 1024 * 1024L;
```

Comment:
Since these are examples we should just use the long primitive instead of the Long object. When our API uses the Long object we should use Long.

Reply:
Done.